packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1359,10 +1355,14 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             }
 
             const trace = Trace.start();
-            const treeWithStats = await this.summarize(fullTree || safe, true /* trackState */);
+            const summarizeResult = await this.summarize(fullTree || safe, true /* trackState */);
+
+            // Run garbage collection on the GC nodes returned by summarize.
+            const startingRoutes = [ "/" ];
+            this.garbageCollector.runGC(summarizeResult.gcNodes, startingRoutes);
```

Comment:
After GC has run, data stores should be notified of routes that are used in it. I will address that separately. Created a work item to track this - #4493